interface/ceed-operator.c

```diff
@@ -779,10 +779,13 @@ int CeedOperatorAssembleLinearQFunction(CeedOperator op, CeedVector *assembled,
 }
 
 /**
-  @brief Assemble the diagonal of a square linear Operator
+  @brief Assemble the diagonal of a square linear CeedOperator
 
   This returns a CeedVector containing the diagonal of a linear CeedOperator.
 
+  Note: Currently only non-composite CeedOperators with a single field are
+          supported.
```

Comment:
Do we test this at present (and error in a sensible way)?

Reply:
Currently? No. But it's a small fix to enable the composite support, so I was going to add that in a pr on Monday

Reply:
Okay, we should still test single-field so people don't get confusing results if they try.